labs/lab12.md

```diff
@@ -227,70 +231,69 @@ There are multiple bombs with different difficulties to make the challenge more
-"""
-BOMB_DISMANTLED_STEP = {bomb.get_wires: cut_wires}
+ - `bomb.get_wires()` - returns a list of colors, e.g. `["blue", "green", "orange"]`
+ - `bomb.cut_XX_wire()` - There is one function per color. We do not provide the full list of colors. You'll need to figure these out through trial and error.
 
-bomb(level=1)
-bomb.console("Start dismantle bomb program.")  # Step one in the bomb dismantling process
+ Note: After cutting all wires succesfully from a single `get_wires` call, you may need to call `get_wires` again. For example, if on the first call of `get_wires` you get `["blue", "green", "orange"]`, the bomb may or may not be dismantled yet. You'll need to check using `is_dismantled`. If it is not yet dismantled, you should call `get_wires` to get the next set of wires.
 
-find_next_step(bomb)  # A recursive function to dismantle all steps
-```
+The set of colors is randomly chosen each time you play. The number of times you'll have to call `get_wires` is randomly chosen each time you play.
+
+For a video demo giving a solution for level 1, see: []().
+
+### Level 2
+
+On level 2, there are three new thing you'll need to consider.
+ - `bomb.get_wires()` - returns a list of colors just like level 1, e.g. ["blue", "green", "orange"], but only if the next step is to cut wires. If the next step is NOT to cut wires or the bomb is dismantled, then a `BombStartsBeeping` exception occurs, which you should catch to keep the game running.
+ - `bomb.get_cipher()` - returns a list of numbers that must be entered into the console, e.g. could return `[3, 1, 5, 9]`, which would require calling `bomb.console(3)`, then `bomb.console(1)`, and so forth. This only happens if the next step is to enter codes into the console. If the next step is NOT to enter numbers into the console or the bomb is dismantled, then a `BombStartsBeeping` execption occurs, which you should catch to keep the game running.
+ - There is no way to ask the bomb directly whether the next step is to cut wires or enter codes. You must use trial and error with `get_wires` and `get_cipher`. Note that once the bomb is dismantled, both `get_wires` and `get_cipher` will cause the `BombStartsBeeping` exception.
+
+Like all other levels, the wires and codes are random each time you play.
 
+For a video demo giving a solution for level 2, see: []().
 
+### Level 3
 
+Same as level 2, except that:
 
+ - `bomb.get_cipher()` - Instead of a list, will now sometimes return a nested list, e.g. `[1, 2, [3, 4], [[20, 30], 7, 8], 9]`, in which case you should now enter the numbers 1, then 2, then 3, then 4, then 20, then 30, then 7, then 8, then 9, in that order. There is a clever way to do this.
 
+### Level 4
 
+Level 4 has these four new functions.
 
+ - `bomb.get_part()` - gets a part to be removed.
+ - `bomb.get_screws()` - gets a list of screws that must be removed. This must be called immediately AFTER `get_part`, but before `remove_part`.
```

Comment:
Let them figure the order, just tell them how to get the screws. The error will tell you that you did not remove the screws.